tensorflow/python/keras/layers/normalization.py

```diff
@@ -281,53 +281,55 @@ def _get_shape_and_axis_for_fused(self, nd_shape, nd_axis):
     this function without affecting the correctness of the computation.
 
     Arguments:
-      nd_shape: List. The original shape of the operation.
+      nd_shape: Tensor. The original shape of the operation.
       nd_axis: Integer. The original axis of the operation.
 
     Returns:
-      shape: List. A 4D shape.
+      shape: Tensor. A 4D shape.
       axis: Integer. An axis (always 1 or 3).
     """
     assert(isinstance(nd_axis, int))
-    ndims = len(nd_shape)
+    ndims = nd_shape.shape[0]
     shape = nd_shape[:]
     axis = nd_shape + nd_axis if nd_axis < 0 else nd_axis
     # First check if the axis needs to be moved.
     if axis not in (1, ndims - 1):
       # Move axis to dim 1.
       if axis == 0:
         # Transform [C, ...] to [1, C, ...].
-        shape.insert(0, 1)
+        shape = array_ops.concat([constant_op.constant([1]), shape], axis=0)
         ndims += 1
       else:
         # Merge excess pre-axis dims into first dim.
         # Transform [N, ..., C, ...] to [product(N, ...), C, ...].
-        product = 1
-        for dim in shape[:axis]:
-          product *= dim
-        shape[:axis] = [product]
+        product = math_ops.reduce_prod(shape[:axis])
```

Comment:
If you pass `keepdims=True` here, I think you don't need to call `array_ops.reshape` on the line below. This makes it slightly simpler.

And same for the place below you call `math_ops.reduce_prod`

Reply:
Done.